dask_expr/tests/test_collection.py

```diff
@@ -1261,3 +1261,48 @@ def test_replace_filtered_combine_similar():
     # Filter expressions (and not the other way around)
     similar = list(df.find_operations(Replace))
     assert all(isinstance(op.frame, Filter) for op in similar)
+
+
+def test_squash_assigns_togeher():
+    # pdf = lib.DataFrame(
+    #     {
+    #         "a": [1, 2, 3, 4],
+    #         "b": 2,
+    #         "c": 3,
+    #         "d": 4,
+    #         "e": 5
+    #     }
+    # )
+    # df = from_pandas(pdf, npartitions=2)
+    #
+    # df["x"] = df.a
+    # df["y"] = df.b
+    # df["z"] = df.c
+    #
+    # df = df[["x", "y", "z", "d"]]
+    # q = df.optimize(fuse=False)
+    # assert len(list(q.find_operations(Assign))) == 1
+    #
+    # pdf["x"] = pdf.a
+    # pdf["y"] = pdf.b
+    # pdf["z"] = pdf.c
+    #
+    # pdf = pdf[["x", "y", "z", "d"]]
+    # assert_eq(pdf, df)
+
+    pdf = lib.DataFrame({"a": [1, 2, 3, 4], "d": 4})
+    df = from_pandas(pdf, npartitions=2)
+
+    # This does not introduce an intermediate projection
+    df["x"] = df.a
+    df["y"] = df.a
```

Comment:
Would this also work if this was `df.a + 1`?

Reply:
Yeah that would work, but I don't know what would happen with df["y"] = df["x"], I'll do more tests later